tests/functional/test_fast_deps.py

```diff
@@ -101,3 +103,31 @@ def test_hash_mismatch(script: PipTestEnvironment, tmp_path: pathlib.Path) -> No
         expect_error=True,
     )
     assert "DO NOT MATCH THE HASHES" in result.stderr
+
+
+@mark.network
+def test_hash_mismatch_existing_download(
+    script: PipTestEnvironment, tmp_path: pathlib.Path
+) -> None:
+    reqs = tmp_path / "requirements.txt"
+    reqs.write_text("idna==2.10")
+    dl_dir = tmp_path / "downloads"
+    dl_dir.mkdir()
+    idna_wheel = dl_dir / "idna-2.10-py2.py3-none-any.whl"
+    idna_wheel.write_text("asdf")
+    result = script.pip(
+        "download",
+        "--use-feature=fast-deps",
```

Comment:
Why does this need fast-deps? I get that this is in `test_fast_deps`, but why does fast-deps need a test for this specifically? Presumably it takes a different code path than the "not fast deps" version, so maybe a comment explaining why we need this would help?

Reply:
Added comment!